frontend/src/components/ClientMaintenanceCheck.tsx

```diff
@@ -0,0 +1,28 @@
+"use client";
+
+import { useFeatureFlags } from "src/hooks/useFeatureFlags";
+
+import { redirect, RedirectType } from "next/navigation";
+import { useEffect } from "react";
+
+/**
+ * View for managing feature flags
```

Comment:
this comment is a little confusing - I'd say this component is designed to redirect away from the maintenance page if all maintenance related feature flags are off